src/index.ts

```diff
@@ -81,7 +81,7 @@ export interface ParseOptions {
    *
    * @default decodeURIComponent
    */
-  decode?: (str: string) => string | undefined;
```

Comment:
It's fine to keep these signatures, if someone wants to overwrite and `return undefined` in decode it'd still work.

Reply:
I can edit this PR if you're OK with it?

Reply:
Go ahead